### PR TITLE
Update From CircleCI to GitHub Actions

### PR DESCRIPTION
**Pull request by positiveVibes4all**
## Summary 
Updated everything needed to move from Circle CI and implemented changesets version bumps similar to Contexture. Updates were made to package.json and scripts were added to facilitate this change.

### Testing
- Changes where tested end to end in a forked repo to ensure deploy works, and the package is usable after an NPM publish.
  - https://www.npmjs.com/package/futil-js-ht
- GitHub Pages deploy was tested in the forked repository as well with the same changes.
  - https://positivevibes4all.github.io/futil-js/#deepMap

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Release or Create Release PR
+        uses: changesets/action@v1
+        id: changesets
+        with:
+          version: yarn version
+          publish: yarn publish
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Publish Docs
```
Comment (stellarhoof) [inline .github/workflows/release.yml:42]: (Not required) You could also make a script and on the `yarn publish` script do `yarn npm publish --tolerate-republish && ./scripts/publish-docs`

**Review by stellarhoof — CHANGES_REQUESTED** [review]

**Comment by stellarhoof** [comment]
@positiveVibes4all Great job with this PR and testing it in a fork!

File: .github/workflows/release.yml (hunk 1 above)
Comment (positiveVibes4all) [reply .github/workflows/release.yml:42]: I had to make some changes to the publish stuff, I will clean up the scripts soon, but for now its best to leave this as is I think.

File: package.json [hunk 2]
```diff
@@ -33,45 +33,65 @@
...
-    "docs-publish": "cd docs/beta && yarn run deploy"
+    "docs-publish": "cd docs/beta && yarn run deploy",
+    "version": "yarn changeset version && yarn install --mode=update-lockfile",
+    "publish": "yarn npm publish --tolerate-republish"
   },
   "author": {
-    "name": "Samuel Greene",
-    "email": "sgreene@smartprocure.us"
+    "name": "hthornley-govspend",
+    "email": "hthornley@govspenc.com"
```
[the remark this answers is not in the document]
Comment (positiveVibes4all) [reply package.json:42]: This was leftover from my forked code, removed it.

**Comment by positiveVibes4all** [comment]
> @positiveVibes4all Great job with this PR and testing it in a fork!
> 
> Do you know why there's still a CircleCI check?

We probably have an app setup, I cannot see the config stuff, we also need a secret on this repo for our NPM token to publish. 

 In the release actions => NPM_TOKEN: '${{ secrets.NPM_AUTH_TOKEN }}'

File: .changeset/real-walls-cheat.md
[the remark this answers is not in the document]
Comment (positiveVibes4all) [reply .changeset/real-walls-cheat.md:1]: The other changes that did not make it were not library specific, so I removed them.

File: package.json [hunk 3]
```diff
@@ -3,9 +3,9 @@
   "version": "1.76.1",
   "description": "F(unctional) util(ities). Resistance is futile.",
   "main": "lib/futil-js.js",
-  "workspaces": [
-    "docs/beta"
-  ],
+  "comments": {
+    "devDependencies": "typescript and eslintconfig dependencies should get removed when making this when refactor happens"
```
Comment (stellarhoof) [inline package.json:7]: What refactor?

**Comment by stellarhoof** [comment]
- We have an npm token called `NPM_TOKEN` (no `AUTH`)
- I gave the changeset bot access to this repo
- We have no circle app on this repo, so still not sure where that check is coming from

File: package.json (hunk 3 above)
Comment (positiveVibes4all) [reply package.json]: updated comment

**Review by stellarhoof — APPROVED** [review]

**Pull request merged**